packages/fx-core/src/question/scaffold/vs/createRootNode.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IQTreeNode, OptionItem, Platform } from "@microsoft/teamsfx-api";
+import { getLocalizedString } from "../../../common/localizeUtils";
+import { QuestionNames } from "../../constants";
+import { appNameQuestion, folderQuestion } from "../../create";
+import { TemplateNames } from "../../templates";
+import {
+  BotCapabilityOptions,
+  CustomCopilotCapabilityOptions,
+  MeCapabilityOptions,
+  setTemplateName,
+} from "../vsc/CapabilityOptions";
+import { folderAndAppNameCondition, languageNode } from "../vsc/createRootNode";
+import { llmServiceNode } from "../vsc/customAgentProjectTypeNode";
+import { daProjectTypeNode } from "../vsc/daProjectTypeNode";
+import { botTriggerNode } from "../vsc/m365ProjectTypeNode";
+
+export class VSCapabilityOptions {
+  // empty
+  static empty(): OptionItem {
+    return {
+      id: "empty",
+      label: "Empty",
+      data: TemplateNames.Empty,
+    };
+  }
+  static declarativeAgent(): OptionItem {
+    return {
+      id: "declarative-agent",
+      label: getLocalizedString("core.createProjectQuestion.projectType.declarativeAgent.label"),
+      detail: getLocalizedString("core.createProjectQuestion.projectType.declarativeAgent.detail"),
+    };
+  }
+  static nonSsoTab(): OptionItem {
+    return {
+      id: "tab-non-sso",
+      label: `${getLocalizedString("core.TabNonSso.label")}`,
+      detail: getLocalizedString("core.TabNonSso.detail"),
+      description: getLocalizedString(
+        "core.createProjectQuestion.option.description.worksInOutlookM365"
+      ),
+      data: TemplateNames.TabSSR,
+    };
+  }
+  static tab(): OptionItem {
+    return {
+      id: "tab",
+      label: getLocalizedString("core.TabOption.label"),
+      description: getLocalizedString("core.TabOption.description"),
+      detail: getLocalizedString("core.TabOption.detail"),
+      data: TemplateNames.SsoTabSSR,
+    };
+  }
+}
+
+/**
+ *
+ * FxCore API for scaffold: scaffold(questionModel: IQTreeNode, generators: DefaultTemplateGenerator[]): Promise<Result<any, FxError>>
+ * Dedicated for VS Code platform
```

Comment:
Typo, for VS. BTW, question model doesn't apply to VS before. Is this something going to change in this refactor?

Reply:
It does apply for VS in the same question model with many `runtime` condition previously. Now VS question model is seperated.